protocols/virtualdefund/virtualdefund.go

```diff
@@ -228,6 +324,41 @@ func (o Objective) signedByMe() bool {
 
 }
 
+// isRightDefunded returns whether the ledger channel ToMyRight has been defunded
+// If ToMyRight==nil then we return true
+func (o Objective) isRightDefunded() bool {
+	if o.ToMyRight == nil {
+		return true
+	}
+
+	included := o.ToMyRight.IncludesTarget(o.VId())
+	return !included
+}
+
+// isLeftDefunded returns whether the ledger channel ToMyLeft has been defunded
+// If ToMyLeft==nil then we return true
+func (o Objective) isLeftDefunded() bool {
+	if o.ToMyLeft == nil {
+		return true
+	}
+
+	included := o.ToMyLeft.IncludesTarget(o.VId())
+	return !included
+}
+
+// createSignedProposalMessage returns a signed proposal message addressed to the counterparty in the given ledger
+func (o *Objective) createSignedProposalMessage(sp consensus_channel.SignedProposal, ledger *consensus_channel.ConsensusChannel) protocols.Message {
```

Comment:
This code ought to be shared. Do we need a `wire` package?

Reply:
If we [share code](https://github.com/statechannels/go-nitro/pull/540#discussion_r857812269) between `virtualfund` and `virtualdefund` we could probably include that there for now.